test/integration/pubsub/test_pubsub_requests.rb

```diff
@@ -12,20 +12,32 @@ def setup
   end
 
   def delete_test_resources
-    topics = @client.list_topics[:body]["topics"]
-    unless topics.nil?
-      topics.
-        map { |t| t["name"] }.
-        select { |t| t.start_with?(topic_resource_prefix) }.
-        each { |t| @client.delete_topic(t) }
+    topics_result = @client.list_topics
+    unless topics_result.topics.nil?
+      begin
+        topics_result.topics.
+          map(&:name).
+          select { |t| t.start_with?(topic_resource_prefix) }.
+          each { |t| @client.delete_topic(t) }
+      # We ignore errors here as they are flaky due to the delay in list operations
+      # representing our operations.
+      rescue Google::Apis::Error
```

Comment:
Are there not specific errors relating to just listing? If not that's fine, it'd just be cool if there was.